contracts/margin/external/exchangewrappers/ZeroExV2MultiOrderExchangeWrapper.sol

```diff
@@ -171,15 +176,56 @@ contract ZeroExV2MultiOrderExchangeWrapper is
         total.takerAmount = 0;
         total.makerAmount = desiredMakerToken;
 
+        // gets the exchange cost. modifies total
+        getExchangeCostInternal(
+            makerToken,
+            orders,
+            total
+        );
+
+        // require that all amount was bought
+        require(
+            total.makerAmount == 0,
+            "ZeroExV2MultiOrderExchangeWrapper#getExchangeCost: Cannot buy enough maker token"
+        );
+
+        // validate that max price will not be violated
+        validateTradePrice(
+            priceRatio,
+            total.takerAmount,
+            desiredMakerToken
+        );
+
+        // return the amount of taker token needed
+        return total.takerAmount;
+    }
+
+    // ============ Private Functions ============
+
+    /**
+     * Gets the amount of takerToken required to fill the amount of total.makerToken.
+     * Does not return a value, only modifies the values inside total.
```

Comment:
generally I don't like the pattern of modifying arguments rather than returning a value. It's hard to know that's happening besides comments all over the place. Is this just necessary because of the stack limit?

Reply:
done